packages/quick-edit/patches/0011-Support-project-wide-typescript-on-web.patch

```diff
@@ -0,0 +1,4012 @@
++	private _lookup(uri: URI, silent: boolean): Entry | undefined;
++	private _lookup(uri: URI, silent: boolean): Entry | undefined {
++		const parts = uri.path.split('/');
++		let entry: Entry = this.root;
++		for (const part of parts) {
++			if (!part) {
++				continue;
++			}
++			let child: Entry | undefined;
++			if (entry instanceof Directory) {
++				child = entry.entries.get(part);
++			}
++			if (!child) {
++				if (!silent) {
++					throw FileSystemError.FileNotFound(uri);
++				} else {
++					return undefined;
++				}
++			}
++			entry = child;
++		}
++		return entry;
++	}
++
++	private _lookupAsDirectory(uri: URI, silent: boolean): Directory {
++		const entry = this._lookup(uri, silent);
++		if (entry instanceof Directory) {
++			return entry;
++		}
++		throw FileSystemError.FileNotADirectory(uri);
++	}
++
++	private _lookupAsFile(uri: URI, silent: boolean): File {
++		const entry = this._lookup(uri, silent);
++		if (entry instanceof File) {
++			return entry;
++		}
++		throw FileSystemError.FileIsADirectory(uri);
++	}
++
++	private _lookupParentDirectory(uri: URI): Directory {
++		const dirname = uri.with({ path: this._dirname(uri.path) });
++		return this._lookupAsDirectory(dirname, false);
++	}
++
++	private _basename(path: string): string {
++		path = this._rtrim(path, '/');
++		if (!path) {
++			return '';
++		}
++
++		return path.substr(path.lastIndexOf('/') + 1);
++	}
++
++	private _dirname(path: string): string {
++		path = this._rtrim(path, '/');
++		if (!path) {
++			return '/';
++		}
++
++		return path.substr(0, path.lastIndexOf('/'));
++	}
++
++	private _rtrim(haystack: string, needle: string): string {
++		if (!haystack || !needle) {
++			return haystack;
++		}
++
++		const needleLen = needle.length,
++			haystackLen = haystack.length;
++
++		if (needleLen === 0 || haystackLen === 0) {
++			return haystack;
++		}
++
++		let offset = haystackLen,
++			idx = -1;
++
++		// eslint-disable-next-line
++		while (true) {
++			idx = haystack.lastIndexOf(needle, offset - 1);
++			if (idx === -1 || idx + needleLen !== offset) {
++				break;
++			}
++			if (idx === 0) {
++				return '';
++			}
++			offset = idx;
++		}
++
++		return haystack.substring(0, offset);
++	}
++}
+diff --git a/extensions/typescript-language-features/web/webServer.ts b/extensions/typescript-language-features/web/webServer.ts
+index 1db00fe..da46091 100644
+--- a/extensions/typescript-language-features/web/webServer.ts
++++ b/extensions/typescript-language-features/web/webServer.ts
+@@ -6,10 +6,14 @@
+ /// <reference lib='webworker' />
+ 
+ import * as ts from 'typescript/lib/tsserverlibrary';
+-import { ApiClient, FileType, Requests } from '@vscode/sync-api-client';
+-import { ClientConnection } from '@vscode/sync-api-common/browser';
+ import { URI } from 'vscode-uri';
+ 
++// Keep an in-memory representation of the current Worker script.
++// This is in-memory within this (Web) Worker to avoid a dependency on SharedArrayBuffer
```

Comment:
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer#security_requirements. In essence, Spectre meant that `SharedArrayBuffer`s were effectively disabled without strict cross-origin headers (which aren't set by the embedder of Quick Edit, for instance—headers would have to be set on essentially every resource requested by the _top_ level, not just Quick Edit). In particular, things loaded from within the Quick Edit preview would fail, since they'd be third party resources without these headers set.